cli/config/configfile/file.go

```diff
@@ -24,28 +24,29 @@ const (
-	CredentialHelpers    map[string]string           `json:"credHelpers,omitempty"`
-	Filename             string                      `json:"-"` // Note: for internal use only
-	ServiceInspectFormat string                      `json:"serviceInspectFormat,omitempty"`
-	ServicesFormat       string                      `json:"servicesFormat,omitempty"`
-	TasksFormat          string                      `json:"tasksFormat,omitempty"`
-	SecretFormat         string                      `json:"secretFormat,omitempty"`
-	ConfigFormat         string                      `json:"configFormat,omitempty"`
-	NodesFormat          string                      `json:"nodesFormat,omitempty"`
-	PruneFilters         []string                    `json:"pruneFilters,omitempty"`
-	Proxies              map[string]ProxyConfig      `json:"proxies,omitempty"`
-	Experimental         string                      `json:"experimental,omitempty"`
-	Orchestrator         string                      `json:"orchestrator,omitempty"`
+	AuthConfigs            map[string]types.AuthConfig `json:"auths"`
+	HTTPHeaders            map[string]string           `json:"HttpHeaders,omitempty"`
+	PsFormat               string                      `json:"psFormat,omitempty"`
+	ImagesFormat           string                      `json:"imagesFormat,omitempty"`
+	NetworksFormat         string                      `json:"networksFormat,omitempty"`
+	PluginsFormat          string                      `json:"pluginsFormat,omitempty"`
+	VolumesFormat          string                      `json:"volumesFormat,omitempty"`
+	StatsFormat            string                      `json:"statsFormat,omitempty"`
+	DetachKeys             string                      `json:"detachKeys,omitempty"`
+	CredentialsStore       string                      `json:"credsStore,omitempty"`
+	CredentialHelpers      map[string]string           `json:"credHelpers,omitempty"`
+	Filename               string                      `json:"-"` // Note: for internal use only
+	ServiceInspectFormat   string                      `json:"serviceInspectFormat,omitempty"`
+	ServicesFormat         string                      `json:"servicesFormat,omitempty"`
+	TasksFormat            string                      `json:"tasksFormat,omitempty"`
+	SecretFormat           string                      `json:"secretFormat,omitempty"`
+	ConfigFormat           string                      `json:"configFormat,omitempty"`
+	NodesFormat            string                      `json:"nodesFormat,omitempty"`
+	PruneFilters           []string                    `json:"pruneFilters,omitempty"`
+	Proxies                map[string]ProxyConfig      `json:"proxies,omitempty"`
+	Experimental           string                      `json:"experimental,omitempty"`
+	Orchestrator           string                      `json:"orchestrator,omitempty"`
+	StackListAllNamespaces string                      `json:"stackListAllNamespaces,omitempty"`
```

Comment:
I wonder if we should start to using some `nested` struct, like 
```
{
  "orchestrator": "kubernetes",
  "stack": {
    "list": {
      "allNamespaces": "disabled",
      "format": "…",
    }
  }
}
```

cc @thaJeztah

Reply:
Why not, but maybe in a follow up?

Reply:
Will we only use this configuration for _stacks_, or also for `docker service ls` ? (wondering about the `Stack` prefix.

Alternatively, if we want to use a boolean instead of the `disabled` string, we could use `{"singleNamespace": true}`

Reply:
Ok so what about
```
{
  "kubernetes": {
    "singleNamespace": true
  }
}
```
And by the way, singleNamespace makes you think you can only list one namespace, but you still can use multiple `--namespace` in the `docker stack ls` command, so it could be a little confusing?

Reply:
@thaJeztah hum good point.
I tend to agree with @silvin-lubecki on the `singleNamespace` name… but nothing really pop in my head for a correct name.

Reply:
SGTM